src/libstore/local-fs-store.hh

```diff
@@ -62,4 +65,24 @@ public:
 
 };
 
+struct IndirectRootStore : public virtual LocalFSStore
+{
+    inline static std::string operationName = "Indirect GC roots registration";
+
+    /* Add an indirect root, which is merely a symlink to `path' from
+       /nix/var/nix/gcroots/auto/<hash of `path'>.  `path' is supposed
+       to be a symlink to a store path.  The garbage collector will
+       automatically remove the indirect root when it finds that
+       `path' has disappeared. */
+    virtual void addIndirectRoot(const Path & path) = 0;
+};
+
+
+struct LocalPermRootStore : public virtual IndirectRootStore
+{
+    inline static std::string operationName = "Local permanent GC roots registration";
+
```

Comment:
You only need this if you need to downcast the store; I don't think you need to all the way to `LocalPermRootStore` just `IndirectRootStore` so this shouldn't be needed?

Reply:
```suggestion
```